### ISSUE TITLE
Kleisli

### ISSUE THREAD

**Issue by DanGrayson** [UniMath/UniMath#1249]
There are two definitions of this term:
```
/Users/dan/src/ProofChecking/UniMath/UniMath.git/UniMath/CategoryTheory/Monads/Kleisli.v
23:   Definition Kleisli
/Users/dan/src/ProofChecking/UniMath/UniMath.git/UniMath/CategoryTheory/Monads/KTriples.v
78: Definition Kleisli
```

I'd like to refer to the formalized form of "monad in Kleisli form" by name in VV's paper on C-systems from a monad, so it would be good to have just the one that looks like this under that name:

<img width="833" alt="Screen Shot 2019-07-10 at 10 27 54 AM" src="https://user-images.githubusercontent.com/700228/60977397-69628280-a2fd-11e9-92ba-7a95d952ed74.png">

**Comment by DanGrayson** [comment]
Actually, shouldn't it be something like "Kleisli_monad" instead of "Kleisli"?  That would be more descriptive.

**Comment by DanGrayson** [comment]
Actually, those two definitions of Kleisli are identical.  The identity would be clearer to casual observers after this change:
```
--- a/UniMath/CategoryTheory/Monads/KTriples.v
+++ b/UniMath/CategoryTheory/Monads/KTriples.v
@@ -22,6 +22,7 @@ Require Import UniMath.CategoryTheory.Core.NaturalTransformations.
 Require Import UniMath.CategoryTheory.whiskering.
 Require Import UniMath.CategoryTheory.Monads.Monads.
 Require Import UniMath.CategoryTheory.Monads.RelativeMonads.
+Require Import UniMath.CategoryTheory.Monads.Kleisli.
 
 Local Open Scope cat.
 
@@ -80,7 +81,7 @@ Definition Kleisli: UU :=
 (* argument [C] will be set as not implicit after the end of the section *)
 Coercion Kleisli_Data_from_Kleisli (T : Kleisli) : Kleisli_Data := pr1 T.
 
-Lemma Kleisli_is_rel_monad: Kleisli = RelMonad (functor_identity C).
+Goal Kleisli = Kleisli.Kleisli C.
 Proof.
   apply idpath.
 Qed.
```

**Comment by DanGrayson** [comment]
Is there a good reason to have two identical definitions of the same identifier in different files?

**Comment by rmatthes** [comment]
When I worked on the Monads directory last year, I was confronted with several competing implementations by several groups of authors. I tried to preserve the individual contributions as proper files and did not set out to do global changes, in particular no name changes. However, Kleisli.Kleisli became just an instance of the relative monads - with the aim to reduce code duplication.

Still, is the concept of monad in the form of a Kleisli triple not better be kept with the spelt-out definitions (data and laws)? Even if it is plainly an instance of relative monads. "Plainly" is not exaggerated since being instance is a matter of type-checking. You would still certainly want to ensure that a reader of VV's paper on C-systems can convince herself/himself that the formalized concept agrees with the paper concept without doing the type identification work the type-checker has no problems with.

Concerning your second last comment, I agree that `Goal` is better than `Lemma` and that `Kleisli.Kleisli` is better than its definiens.

I also agree that `Kleisli` is not a proper mathematical identifier. Renaming it like you suggest should be followed by renaming of the lemmas that already have long names.

I have to mention that I am busy until Monday but I absolutely wanted to give a first answer in a timely manner.
